### PR TITLE
[SR-7040] Redundant prefix of compilation flag specific error

### PR DESCRIPTION
**Pull request by Nonchalant**
Xcode (and possibly other IDEs) provide a UI over -D flags where you provide the flag name and -D is prepended for you. Users might reasonably add a redundant -D themselves, leading to the invalid flag `-D-DFOO` getting passed into the compiler. The driver could detect this and raise a specific error message alerting the user to their mistake.

Resolves [SR-7040](https://bugs.swift.org/browse/SR-7040).

### CODE REVIEW

File: include/swift/AST/DiagnosticsFrontend.def [hunk 1]
```diff
@@ -209,6 +209,10 @@ ERROR(symbol_in_ir_not_in_tbd,none,
       "symbol '%0' (%1) is in generated IR file, but not in TBD file",
       (StringRef, StringRef))
 
+ERROR(redundant_prefix_compilation_flag,none,
+      "did you provide a redundant '-D' in your build settings? (got '-D%0')",
```
Comment (jrose-apple) [inline include/swift/AST/DiagnosticsFrontend.def]: We usually try to say what went wrong first and then offer a suggestion, so maybe something more like "invalid argument '-D%0'; did you…"?

**Review by jrose-apple — COMMENTED** [review]
Thanks for taking this on! I have a suggestion for the phrasing of the message, but the implementation looks good and I'm glad it worked so cleanly.

File: include/swift/AST/DiagnosticsFrontend.def (hunk 1 above)
Comment (Nonchalant) [reply include/swift/AST/DiagnosticsFrontend.def]: Thank you! I will try to fix message!! ✨

**Comment by Nonchalant** [comment]
@swift-ci please test

**Comment by Nonchalant** [comment]
@swift-ci please smoke test

**Review by jrose-apple — APPROVED** [review]

**Comment by jrose-apple** [comment]
Looks great!

@swift-ci Please smoke test

**Comment by rintaro** [comment]
Disabled failed test in https://github.com/apple/swift/pull/15005

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Pull request merged**